### PR TITLE
use chrome devtools protocol directly

### PR DESCRIPTION
**Pull request by cscheid**
This replaces our use of `puppeteer` with a minimal, self-contained deno port of `chrome-remote-interface`. This fixes our Linux issues with `puppeteer` in my Ubuntu 22.04 tests.

### CODE REVIEW

File: src/core/cri/cri.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      await client.close();
+      return result;
+    } catch (e) {
+      await client.close();
+      throw e;
+    }
+  });
+}
+
+export async function criClient(appPath?: string, port = 9222) {
```
Comment (jjallaire) [inline src/core/cri/cri.ts:56]: Do we need to find a new open port here? (or gracefully handle finding a different port if this one is taken?). We have some tools for this in core

File: src/core/cri/deno-cri-test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { criClient } from "./cri.ts";
+
+const m = await criClient();
+await m.open("https://github.com");
```
Comment (jjallaire) [inline src/core/cri/deno-cri-test.ts]: Eliminate this file

File: src/core/cri/deno-cri/devtools.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ * devtools.js
+ *
+ * Copyright (c) 2021 Andrea Cardaci <cyrus.and@gmail.com>
+ *
+ * Deno port Copyright (C) 2022 by RStudio, PBC
+ */
+
+import * as defaults from "./defaults.js";
+import externalRequest from "./external-request.js";
+import localDescriptor from "./protocol.json" assert { type: "json" };
```
Comment (jjallaire) [inline src/core/cri/deno-cri/devtools.js:11]: Load this from resources

**Review by jjallaire — CHANGES_REQUESTED** [review]

File: src/core/cri/deno-cri-test.ts (hunk 2 above)
Comment (cscheid) [reply src/core/cri/deno-cri-test.ts]: Done

File: src/core/cri/deno-cri/devtools.js (hunk 3 above)
Comment (cscheid) [reply src/core/cri/deno-cri/devtools.js:11]: Done.

File: src/core/cri/cri.ts (hunk 1 above)
Comment (cscheid) [reply src/core/cri/cri.ts]: Done.

**Pull request merged**